### PR TITLE
fix(paramserver): validates yaml or json data

### PR DESCRIPTION
**Pull request by pallabpain**
### Description
Validates the JSON or YAML data if the file is not uploaded as a blob before invoking the create_file API.

demo

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -10,6 +10,7 @@ urllib3 = ">=1.23"
 python-dateutil = ">=2.8.1"
 pytz = "*"
 jsonschema = "==4.0.0"
+pyyaml = ">=5.4.1"
```
Comment (ankitrgadiya) [inline Pipfile:13]: todo: Please add this in the setup.py file as well.

File: rapyuta_io/clients/paramserver.py [hunk 2]
```diff
@@ -142,10 +146,14 @@ def process_dir(self, executor, rootdir, tree_path, level, dir_futures, file_fut
                 if file_name.endswith('.yaml'):
                     with open(full_path, 'r') as f:
                         data = f.read()
+                    if not is_valid_yaml(data):
+                        raise InvalidYAMLError(full_path)
```
[the remark this answers is not in the document]
Comment (pallabpain) [reply rapyuta_io/clients/paramserver.py:150]: We'll lose the path context if we do this. Instead, I changed the methods to parse instead. The methods will parse and return data or raise an exception.

File: Pipfile (hunk 1 above)
Comment (pallabpain) [reply Pipfile:13]: Its already added.

**Review by ankitrgadiya — APPROVED** [review]
Looks good

**Pull request merged**